tests/test_database.py

```diff
@@ -1837,6 +1847,24 @@ def test_lookups(self):
 
         self.assertEqual(cm.exception.args[0], 0x41)
 
+    def test_missing_dbc_specifics(self):
+        class NodeMock(object):
+            def __init__(self, name, comment):
+                self.name = name
+                self.comment = comment
+                self.dbc = None
+        
+        nodes = []
+        node = NodeMock("FakeNode", "Comment")
+        nodes.append(node)
+        db = cantools.db.Database(nodes=nodes)
+        signals = []
+        signals.append(cantools.db.Signal("C", 0, 8))
+        db.messages.append(cantools.db.Message(0x20, "D", 8, signals))       
+
+        # Test if dump executes without raising
+        db.as_dbc_string()
+
```

Comment:
I suggest first creating an empty database and then add the node and message, or is there a reason to give the node as argument to the database creation, and then add the message after the database has been created?

Add `from .node import Node` in `cantools/db/__init__.py` to use it as below.

``` python
def test_missing_dbc_specifics(self):
    db = cantools.db.Database()

    node = cantools.db.Node('FakeNode', 'Comment')
    db.nodes.append(node)

    signals = [cantools.db.Signal('C', 0, 8)]
    message = cantools.db.Message(0x20, 'D', 8, signals)
    db.messages.append(message)       

    # Test that dump executes without raising.
    db.as_dbc_string()
```

Reply:
Yes, that was what I was missing...